### PR TITLE
CONSOLE-2371: [i18n] Externalize strings in olm root components

### PR DESCRIPTION
**Pull request by TheRealJon**
Externalize strings in files at root level of `frontend/packages/operator-lifecycle-manager/src/components`

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx:615]: It would be better to make this a separate component outside of this one instead of an inline component so it doesn't remount on every render.

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx:616]: Better to make this a separate component instead of inline

File: frontend/packages/operator-lifecycle-manager/src/components/create-catalog-source.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/create-catalog-source.tsx:95]:
```suggestion
          {t('olm~Create a CatalogSource in order to make operators available in OperatorHub.')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:154]:
```suggestion
        'olm~InstallPlans are created automatically by subscriptions or manually using the CLI.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:239]:
```suggestion
      title={t('olm~InstallPlans')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:258]:
```suggestion
                'olm~Inspect the requirements for the components specified in this InstallPlan before approving.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:272]:
```suggestion
        <SectionHeading text={t('olm~InstallPlan details')} />
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:365]:
```suggestion
          <HintBlock title={t('olm~Review manual InstallPlan')}>
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:443]:
```suggestion
        title={t('olm~No components resolved')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:444]:
```suggestion
        detail={t('olm~This InstallPlan has not been fully resolved yet.')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/k8s-resource.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/k8s-resource.tsx:112]:
```suggestion
          title={t('olm~No resources found')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-group.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-group.tsx:32]:
```suggestion
            'olm~The Operator Lifecycle Manager will not watch this Namespace because it is not configured with an OperatorGroup.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-group.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-group.tsx:58]: Externalize `OperatorGroup` as part of the message instead of using model.plural

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:282]:
```suggestion
          title={t('olm~No Subscriptions found')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:366]:
```suggestion
              'olm~The CatalogSource for this Operator has been removed. The CatalogSource must be added back in order for this Operator to receive any updates.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:380]:
```suggestion
        <SectionHeading text={t('olm~Subscription details')} />
```

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:413]: Externalize `CatalogSource`

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:439]: Externalize `InstallPlan`

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:587]:
```suggestion
                <span className="text-muted">{t('olm~CatalogSource was removed')}</span>
```

**Review by rhamilto — APPROVED** [review]
Lots of great code improvements!

/lgtm

/hold for approvals

/assign @yapei @ahardin-rh @sferich888

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:126]: Screen Shot 2020-12-15 at 6 29 10 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:52]: Screen Shot 2020-12-15 at 6 34 29 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:83]: The string is `Review Manual Install Plan`
Screen Shot 2020-12-15 at 6 38 51 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:35]: Screen Shot 2020-12-15 at 6 44 45 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:20]: should `Succeeded` or `Failed` status be translated? 
Screen Shot 2020-12-15 at 6 47 43 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:19]: should message and 'Unstall' button in uninstall modal be translated? 
Screen Shot 2020-12-15 at 6 50 23 PM

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:441]: disable catalog source modal message and button are not completely translated
Screen Shot 2020-12-15 at 6 58 08 PM

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:129]: CatalogSource Details heading is not translated
Screen Shot 2020-12-15 at 6 56 07 PM

Comment (yapei) [reply frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:129]: PackageManifest details heading is not translated
Screen Shot 2020-12-15 at 6 55 09 PM

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/locales/en/olm.json:20]: We can't localize dynamic k8s resource properties. The "Succeeded" and "Failed" statuses that you see in that column come the CSV `status.phase` which we can't localize.  The "Cannot update" and "CatalogSource was removed" statuses are a static strings that are defined in our code, so we can localize those. I will address localization of the  second line where possible.

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/locales/en/olm.json:19]: This is addressed in #7499

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:441]: See #7499

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:129]: These are actually using the DefaultDetailsPage component, which is outside the scope of this PR.

**Comment by TheRealJon** [comment]
@yapei I've addressed the comments that I could and I think this is ready for another review.

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (yapei) [reply frontend/packages/operator-lifecycle-manager/locales/en/olm.json:20]: Ok, got it

**Comment by yapei** [comment]
thank you @TheRealJon  I've tested again and no more issues found

**Comment by rhamilto** [comment]
@TheRealJon rebase required.

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:33]: I would avoid embedding logic into the message itself using `Trans`. Probably we want two separate complete messages, one when all namespaces is selected and another when a specific project is selected.

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx:828]:
```suggestion
          {t('olm~Create a {{initializationResourceKind}} instance to use this Operator.', {
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:303]: I would use `t('olm~CatalogSources')`

File: frontend/packages/operator-lifecycle-manager/src/components/package-manifest.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/package-manifest.tsx:48]: Avoid passing `t` to a function. Better to call `useTranslation` in the case of component or just use the global instance outside of a component.

File: frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx:346]:
```suggestion
    t('olm~InstallPlan failed');
```

**Comment by TheRealJon** [comment]
@spadgett I've addressed your comments

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:128]: It looks like we have another message with logic in it. We should try to clean that up.

**Comment by TheRealJon** [comment]
@spadgett  rebased. Needs lgtm again.

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:202]: Is there a reason this change was needed? I worry this might cause the component to remount every render since it's now an anonymous component.

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:200]: Probably we should externalize this

```suggestion
      aria-label={t('olm~InstallPlans')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:214]:
```suggestion
              'olm~The Operator has installed successfully. Create the required custom resource to be able to use this operator.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:247]:
```suggestion
      <h2 className="co-clusterserviceversion-install__heading">{t('olm~Installing Operator')}</h2>
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:254]:
```suggestion
        {t('olm~The Operator is being installed. This may take a few minutes.')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:259]:
```suggestion
              'olm~Once the Operator is installed the required custom resource will be available for creation.',
```

**Review by spadgett — APPROVED** [review]
/lgtm

Thanks @TheRealJon !

**Pull request merged**
